spark/src/main/scala/com/acervera/osm4scala/spark/OsmPbfRowIterator.scala

```diff
@@ -88,18 +88,57 @@ object OsmPbfRowIterator {
       case fieldName            => throw new Exception(s"Field $fieldName not valid for Info.")
     })
 
-    private def populateWay(entity: WayEntity, structType: StructType): Seq[Any] = structType.fieldNames.map {
-      case FIELD_ID        => entity.id
-      case FIELD_TYPE      => ENTITY_TYPE_WAY
-      case FIELD_LATITUDE  => null
-      case FIELD_LONGITUDE => null
-      case FIELD_NODES     => UnsafeArrayData.fromPrimitiveArray(entity.nodes.toArray)
-      case FIELD_RELATIONS => new GenericArrayData(Seq.empty)
-      case FIELD_TAGS      => calculateTags(entity.tags)
-      case FIELD_INFO      => entity.info.map(populateInfo).orNull
-      case fieldName       => throw new Exception(s"Field $fieldName not valid for a Way.")
+    private def populateWay(entity: WayEntity, structType: StructType): Seq[Any] = structType.fields.map(f =>
```

Comment:
Sorry, I don't understand this change. Maybe I'm missing something. Why do you prefer `.fields` over `.fieldNames` when the only use of `field` is to get the name so we are adding an extra step in the code? At the end it is exactly the same, but, in my opinion, removing extra step is more readable.

Reply:
It's the same as for Relations, the type of the field is needed